### ISSUE TITLE
Unable to compile

### ISSUE THREAD

**Issue by gpad** [kanatohodets/ricor_ex_talk_prep#1]
Hi @kanatohodets I try to compile your example but it doesn't work and I receive this error:

```
==> basho_poolboy (compile)
Unchecked dependencies for environment dev:
* basho_poolboy (Hex package)
  could not find an app file at _build/dev/lib/basho_poolboy/ebin/basho_poolboy.app. This may happen if the dependency was not yet compiled, or you specified the wrong application name in your deps, or the dependency indeed has no app file (then you can pass app: false as option)
** (Mix) Can't continue due to errors on dependencies
```

I use erlang 18.3 e elixir 1.2.6.

Can you help me please?

**Comment by kanatohodets** [comment]
hi @gpad: please try again with c568b52ef8e and Elixir 1.3 -- I still had an old work-around for basho_poolboy present, which is no longer needed with Elixir 1.3. I had to pin riak_core_ng for now because of some wonky dep stuff, but I'll check with @Licenser to get that sorted.

**Comment by kanatohodets** [comment]
@Licenser: if I use the latest revision of `riak_core_ng` I get this error when fetching deps:

```
Running dependency resolution

Failed to use "lager" (version 3.0.2) because
  basho_exometer_core (version 1.0.0) requires ~>2.1.0
  cuttlefish (version 2.0.7) requires 3.0.2
  deps/riak_core/rebar.config requires ~>3.0.2
  riak_ensemble (version 2.1.3) requires 3.0.2
  riak_sysmon (version 2.1.2) requires 3.0.2
  Locked to 3.0.2 in your mix.lock

** (Mix) Hex dependency resolution failed, relax the version requirements of your dependencies or unlock them (by using mix deps.update or mix deps.unlock)
```

Seems likely to be related to https://github.com/project-fifo/riak_core/commit/fc52ebf51483cdfdfe0c683b609bbfb9de3a0146 -- I know you're not a mix user, so please ping me here or in #riak_core with what I can do to help hunt (or who you think I should chase :)

**Comment by gpad** [comment]
@kanatohodets I try but now `chash` doesnt' compile with erlang 19, required by elixir 1.3. I get this error:

```
==> chash (compile)
src/chash.erl:226: random:uniform/1: the 'random' module is deprecated; use the 'rand' module instead
Compiling src/chash.erl failed:
ERROR: compile failed while processing /home/gpad/workspace/riak/elixir_riak_core_ping/deps/chash: rebar_abort
** (Mix) Could not compile dependency :chash, "/home/gpad/.mix/rebar compile skip_deps=true deps_dir="/home/gpad/workspace/riak/elixir_riak_core_ping/_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile chash", update it with "mix deps.update chash" or clean it with "mix deps.clean chash"
```

instead trying with erlang 18.3 and elixir 1.2.6 (as my first experiment) I get this:

```
* Getting riak_core (https://github.com/project-fifo/riak_core.git)
Cloning into '/home/gpad/workspace/riak/elixir_riak_core_ping/deps/riak_core'...
remote: Counting objects: 16406, done.        
remote: Total 16406 (delta 0), reused 0 (delta 0), pack-reused 16406        
Receiving objects: 100% (16406/16406), 10.26 MiB | 798.00 KiB/s, done.
Resolving deltas: 100% (10237/10237), done.
Checking connectivity... done.
** (FunctionClauseError) no function clause matching in String.split/3
    (elixir) lib/string.ex:334: String.split(~r/0.8.1-p3/, " ", [trim: true])
    (hex) lib/hex/version.ex:108: Hex.Version.custom_requirement/1
    (hex) lib/hex/version.ex:83: Hex.Version.cache/2
    (hex) lib/hex/version.ex:70: Hex.Version.parse_requirement!/1
    (hex) lib/hex/version.ex:22: anonymous fn/2 in Hex.Version.match?/2
    (hex) lib/hex/version.ex:83: Hex.Version.cache/2
    (hex) lib/hex/remote_converger.ex:200: anonymous fn/2 in Hex.RemoteConverger.prepare_locked/3
    (elixir) lib/enum.ex:758: anonymous fn/4 in Enum.filter_map/3
```

How do you compile this experiment? with which versions?

I'm talking about this problem also in elixir forum: https://elixirforum.com/t/error-compiling-erlang-package-with-rebar/981

**Comment by kanatohodets** [comment]
oh, rats, I had compiled Elixir 1.3 from master, so I was still using Erlang 18.

The second error you got:

```
(FunctionClauseError) no function clause matching in String.split/3
```

was fixed in Elixir 1.3. However, the upgrade to Erlang 19 might indeed be a challenge for a bunch of libraries...

I'll spend some more time with this tonight to try to see how challenging it'll be to get Riak Core to work under Erlang 19.

**Comment by kanatohodets** [comment]
Hi @gpad -- a new version of riak_core_ng was released on hex, so with Elixir 1.3+ and Erlang 18, the current mix file in this repo builds and runs cleanly on my machine.

Unfortunately some of the riak core dependencies do not build under Erlang 19, so for now to use Elixir + Riak Core it needs to be on top of Erlang 18.

I'll add a note to the README in this and my other riak core + Elixir repos that Erlang 18 is required.

**Comment by gpad** [comment]
Hi @kanatohodets thank you very much. It's work very well just a reminder, if you copy from hex.pm `{:riak_core, "~> 2.2", hex: :riak_core_ng}` you obtain the version 2.2.5 wiht the bug. If you want the `2.2.6` I need to specify explicitly with `{:riak_core, "~> 2.2.6", hex: :riak_core_ng}`.

thanks again
